### ISSUE TITLE
SoundEntries DBC writing broken.

### ISSUE THREAD

**Issue by kaytotes** [tswow/tswow#345]
**Describe the bug**

The `File` and `Freq` columns should be an array of 10 actual columns in the DBC. Looks like TSWoW is accepting a string instead and then garbling it into an array. 

**To Reproduce**

```ts
console.log(DBC.SoundEntries.find({ ID: 8581 }).objectify());

/**
 * Audio.
 */
export const CUSTOM_AGGRO_SOUND = DBC.SoundEntries.add(20000, {
    SoundType: 10,
    Name: 'Aggro1',
    File: 'Aggro1.wav',
    Freq: 1,
    DirectoryBase: 'Sound\\Creature\\CustomCreatureName',
    Volumefloat: 1.0,
    Flags: 0,
    MinDistance: 100,
    DistanceCutoff: 200,
});

console.log(CUSTOM_AGGRO_SOUND.objectify());
```

Output: 

```
{
  ID: 8581,
  SoundType: 10,
  Name: 'A_CthunYouAreAlready',
  File: [ 'CThunYouAreAlready.wav', '', '', '', '', '', '', '', '', '' ],
  Freq: [
    1, 1, 1, 1, 1,
    1, 1, 1, 1, 1
  ],
  DirectoryBase: 'Sound\\Creature\\CThun',
  Volumefloat: 1,
  Flags: 0,
  MinDistance: 100,
  DistanceCutoff: 200,
  EAXDef: 0,
  SoundEntriesAdvancedID: 0
}
{
  ID: 20000,
  SoundType: 10,
  Name: 'Aggro1',
  File: [
    'A', 'g', 'g', 'r',
    'o', '1', '.', 'w',
    'a', 'v'
  ],
  Freq: [
    0, 0, 0, 0, 0,
    0, 0, 0, 0, 0
  ],
  DirectoryBase: 'Sound\\Creature\\CustomCreatureName',
  Volumefloat: 1,
  Flags: 0,
  MinDistance: 100,
  DistanceCutoff: 200,
  EAXDef: 2,
  SoundEntriesAdvancedID: 0
}
```

I also attempted working around with

```ts
let sounds: Array<string> = ['Aggro1.wav'];

CUSTOM_AGGRO_SOUND.File.set(sounds);
```

But this also produces an error. 

```
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
```

**Expected behavior**
Be able to properly add sound entries via an array.

**Additional Context**
Add any other context or screenshots about the issue here.

**Comment by tswow-admin** [comment]
> **Describe the bug**
> ```ts
> console.log(DBC.SoundEntries.find({ ID: 8581 }).objectify());
> 
> /**
>  * Audio.
>  */
> export const CUSTOM_AGGRO_SOUND = DBC.SoundEntries.add(20000, {
>     SoundType: 10,
>     Name: 'Aggro1',
>     File: 'Aggro1.wav',
>     Freq: 1,
>     DirectoryBase: 'Sound\\Creature\\CustomCreatureName',
>     Volumefloat: 1.0,
>     Flags: 0,
>     MinDistance: 100,
>     DistanceCutoff: 200,
> });
> 
> console.log(CUSTOM_AGGRO_SOUND.objectify());
> ```


Confirmed issue on 8e3814aa6cf34785b1e6ed409eb51762ee55b997 (not yet public)

> I also attempted working around with
> 
> ```ts
> let sounds: Array<string> = ['Aggro1.wav'];
> 
> CUSTOM_AGGRO_SOUND.File.set(sounds);
> ```

Fixed for 0.13. You can work around this by giving it exactly 10 entries on 0.12

**Comment by tswow-admin** [comment]
Complete fix is coming with 0.13

**Comment by kaytotes** [comment]
Just in case anyone else needs this before 0.13 here is the hacky workaround I've been using based on the above info. 

```ts
/**
 * This function should only exist until TS-WoW 0.13 is released and https://github.com/tswow/tswow/issues/345 is closed.
 */
export function AddSoundEntry(
        id: number, name: string, soundType: number, baseDirectory: string, files: Array<string>, frequencies: Array<number> = [1], volume: number = 1.0, minDistance: number = 100, distanceCutoff: number = 200, flags: number = 0
    ) : SoundEntriesRow {
    
    let entry = DBC.SoundEntries.add(id, {
        Name: name,
        SoundType: soundType,
        DirectoryBase: baseDirectory,
        File: '0000000000',
        Freq: 0,
        Volumefloat: volume,
        Flags: flags,
        MinDistance: minDistance,
        DistanceCutoff: distanceCutoff,
    });

    while (files.length < 10) {
        files.push('');
    }

    while (frequencies.length < 10) {
        frequencies.push(0);
    }

    entry.File.set(files);
    entry.Freq.set(frequencies);

    return entry;
}
```
